arborist/server.go

```diff
@@ -238,25 +246,41 @@ func (server *Server) handleAuthProxy(w http.ResponseWriter, r *http.Request) {
 
 	w.Header().Set("REMOTE_USER", info.username)
 
-	rv, err := authorize(&AuthRequest{
-		info.username,
-		info.policies,
-		resourcePath,
-		service,
-		method,
-		server.stmts,
-	})
-	if err != nil {
-		msg := fmt.Sprintf("could not authorize: %s", err.Error())
-		server.logger.Info("tried to handle auth request but input was invalid: %s", msg)
-		response := newErrorResponse(msg, 400, nil)
-		_ = response.write(w, r)
+	var authRequest = AuthRequest{
+		Username: info.username,
+		ClientID: info.clientID,
+		Policies: info.policies,
+		Resource: resourcePath,
+		Service:  service,
+		Method:   method,
+		stmts:    server.stmts,
+	}
+
+	handle := func(rv *AuthResponse, err error) bool {
```

Comment:
I think this could just be inlined, right?

Reply:
Did you mean `handle` please? It was called twice, inlining means copying the code?

Reply:
```go
	var rv bool
	var err error
	if authRequest.ClientID != "" {
		rv, err = authorizeClient(&authRequest)
	} else {
		rv, err = authorizeUser(&authRequest)
	}
	// inline `handle` here
```

Reply:
oh wait, I remembered why - `handle()` was actually called twice in a row sometimes because the `return` is not always executed:

```go
	rv, err := authorizeUser(&authRequest)
	if handle(rv, err) {
		return
	}

	if authRequest.ClientID != "" {
		rv, err = authorizeClient(&authRequest)
		handle(rv, err)
	}
```

Reply:
```go
	rv, err := authorizeUser(&authRequest)
	if !rv.Auth || err != nil {
		// handle err/return early
	}
	if authRequest.ClientID != "" {
		rv, err = authorizeClient(&authRequest)
	}
	// inline `handle` here
```

Reply:
@rudyardrichter thanks for the reminder! Changing to this:

```python
	rv, err := authorizeUser(&authRequest)
	if rv.Auth && err == nil && authRequest.ClientID != ""{
		rv, err = authorizeClient(&authRequest)
	}
	// inline `handle` here
```